odoo/models.py

```diff
@@ -4418,43 +4418,45 @@ def _flush_search(self, domain, fields=None, order=None, seen=None):
         to_flush = defaultdict(set)             # {model_name: field_names}
         if fields:
             to_flush[self._name].update(fields)
-        # also take into account the fields in the record rules
-        domain = list(domain) + (self.env['ir.rule']._compute_domain(self._name, 'read') or [])
-        for arg in domain:
-            if isinstance(arg, str):
-                continue
-            if not isinstance(arg[0], str):
-                continue
-            model_name = self._name
-            for fname in arg[0].split('.'):
-                field = self.env[model_name]._fields.get(fname)
+
+        def process_domain(model, domain):
+            for arg in domain:
+                if isinstance(arg, str):
+                    continue
+                if not isinstance(arg[0], str):
+                    continue
+                comodel = process_path(model, arg[0])
+                if arg[1] in ('child_of', 'parent_of') and comodel._parent_store:
```

Comment:
`comodel._parent_store` should I read it as "is the parent of the comodel stored ?"